File: app/src/main/java/org/deiverbum/app/data/entity/relation/MisaWithComentarios.java
package org.deiverbum.app.data.entity.relation;

import androidx.room.Embedded;
import androidx.room.Relation;

import org.deiverbum.app.data.entity.BibleHomilyJoinEntity;
import org.deiverbum.app.data.entity.BibleReadingEntity;
import org.deiverbum.app.data.entity.MassReadingEntity;
import org.deiverbum.app.model.BibleComment;
import org.deiverbum.app.model.Liturgy;
import org.deiverbum.app.model.MassReading;

import java.util.ArrayList;
import java.util.List;

/**
 * @author A. Cedano
 * @version 1.0
 * @since 2023.1
 */
@SuppressWarnings("SameReturnValue")
public class MisaWithComentarios {
    @Embedded
    /*public LiturgyGroupEntity misaLectura;

    @Relation(
            parentColumn = "groupID",
            entityColumn = "groupFK",
            entity = MassReadingEntity.class
    )*/

    public MassReadingEntity misaLectura;

    @Relation(
            parentColumn = "readingFK",
            entityColumn = "readingFK",
            entity = BibleHomilyJoinEntity.class
    )
    public List<BibleHomilyWithAll> lectura;


    @Relation(
            parentColumn = "readingFK",
            entityColumn = "readingID",
            entity = BibleReadingEntity.class
    )
    public BibleReadingEntity lecturaEntity;

public MassReading getBiblicaMisa() {
        MassReading theModel = lecturaEntity.getDomainModelMisa();
        theModel.setTema(misaLectura.getTema());
        theModel.setOrden(misaLectura.getOrden());
        return theModel;
    }

    public Liturgy getLiturgia() {
        return null;//liturgia.getDomainModel();
    }

    public List<BibleComment> getDomainModel() {
        List<BibleComment> listModel = new ArrayList<>();
        //lectura.
        if (lectura.size() > 0) {
            for (BibleHomilyWithAll item : lectura) {
                //if(homilias.)
                //item.getDomainModel().getOrden();
                BibleComment theModel = item.getDomainModel();
                MassReading biblica = getBiblicaMisa();//misaLectura.getDomainModel();
                biblica.setOrden(misaLectura.getOrden());
                theModel.setBiblica(biblica);
                listModel.add(theModel);
            }
        }
        return listModel;
    }


}

File: app/src/main/java/org/deiverbum/app/data/entity/relation/MisaWithComentarios.kt
package org.deiverbum.app.data.entity.relation

import androidx.room.Embedded
import androidx.room.Relation
import org.deiverbum.app.data.entity.BibleHomilyJoinEntity
import org.deiverbum.app.data.entity.BibleReadingEntity
import org.deiverbum.app.data.entity.MassReadingEntity
import org.deiverbum.app.model.BibleComment
import org.deiverbum.app.model.MassReading

/**
 * @author A. Cedano
 * @version 1.0
 * @since 2023.1
 */
data class MisaWithComentarios(
    @Embedded
    val misaLectura: MassReadingEntity,

    @Relation(
        parentColumn = "readingFK",
        entityColumn = "readingFK",
        entity = BibleHomilyJoinEntity::class
    )
    val lectura: List<BibleHomilyWithAll>,

    @Relation(
        parentColumn = "readingFK",
        entityColumn = "readingID",
        entity = BibleReadingEntity::class
    )
    val lecturaOne: BibleReadingWithBook
) {
    private val biblicaMisa: MassReading
        get() {
            val theModel = lecturaOne.domainModelMisa
            theModel.tema = misaLectura.tema
            theModel.setOrden(misaLectura.orden)
            return theModel
        }
    val domainModel: MutableList<BibleComment>
        get() {
            val listModel: MutableList<BibleComment> = ArrayList()
            if (lectura.isNotEmpty()) {
                for (item in lectura) {
                    val theModel = item.domainModel
                    val biblica = biblicaMisa
                    biblica.setOrden(misaLectura.orden)
                    theModel.biblica = biblica
                    listModel.add(theModel)
                }
            }
            return listModel
        }
}